flake.lock: Update

yannham opened the pull request:
Automated changes by the [update-flake-lock](https://github.com/DeterminateSystems/update-flake-lock) GitHub Action.

```
Flake lock file updates:

• Updated input 'crane':
    'github:ipetkov/crane/6849911446e18e520970cc6b7a691e64ee90d649' (2023-11-09)
  → 'github:ipetkov/crane/27025ab71bdca30e7ed0a16c88fd74c5970fc7f5' (2024-05-09)
• Updated input 'flake-utils':
    'github:numtide/flake-utils/ff7b65b44d01cf9ba6a71320833626af21126384' (2023-09-12)
  → 'github:numtide/flake-utils/b1d9ab70662946ef0850d488da1c9019f3a9752a' (2024-03-11)
• Updated input 'nix-input':
    'github:nixos/nix/1d86bb4f70ee5c2d06810f21bf7cd057ed46712c' (2023-11-17)
  → 'github:nixos/nix/87ab3c0ea4e6f85e7b902050365bb75cf2836fbb' (2024-05-10)
• Added input 'nix-input/flake-parts':
    'github:hercules-ci/flake-parts/9126214d0a59633752a136528f5f3b9aa8565b7d' (2024-04-01)
• Added input 'nix-input/flake-parts/nixpkgs-lib':
    follows 'nix-input/nixpkgs'
• Added input 'nix-input/libgit2':
    'github:libgit2/libgit2/45fd9ed7ae1a9b74b957ef4f337bc3c8b3df01b5' (2023-10-18)
• Removed input 'nix-input/lowdown-src'
• Updated input 'nix-input/nixpkgs':
    'github:NixOS/nixpkgs/9eb24edd6a0027fed010ccfe300a9734d029983c' (2023-11-01)
  → 'github:NixOS/nixpkgs/b550fe4b4776908ac2a861124307045f8e717c8e' (2024-02-28)
• Added input 'nix-input/pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/40e6053ecb65fcbf12863338a6dcefb3f55f1bf8' (2024-04-12)
• Added input 'nix-input/pre-commit-hooks/flake-compat':
    follows 'nix-input'
• Added input 'nix-input/pre-commit-hooks/flake-utils':
    'github:numtide/flake-utils/5aed5285a952e0b949eb3ba02c12fa4fcfef535f' (2022-11-02)
• Added input 'nix-input/pre-commit-hooks/gitignore':
    follows 'nix-input'
• Added input 'nix-input/pre-commit-hooks/nixpkgs':
    follows 'nix-input/nixpkgs'
• Added input 'nix-input/pre-commit-hooks/nixpkgs-stable':
    follows 'nix-input/nixpkgs'
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/bf744fe90419885eefced41b3e5ae442d732712d' (2023-11-14)
  → 'github:NixOS/nixpkgs/f1010e0469db743d14519a1efd37e23f8513d714' (2024-05-09)
• Updated input 'pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/e558068cba67b23b4fbc5537173dbb43748a17e8' (2023-11-15)
  → 'github:cachix/pre-commit-hooks.nix/2849da033884f54822af194400f8dff435ada242' (2024-04-30)
• Updated input 'pre-commit-hooks/flake-compat':
    'github:edolstra/flake-compat/35bb57c0c8d8b62bbfd284272c928ceb64ddbde9' (2023-01-17)
  → 'github:edolstra/flake-compat/0f9255e01c2351cc7d116c072cb317785dd33b33' (2023-10-04)
• Updated input 'pre-commit-hooks/gitignore':
    'github:hercules-ci/gitignore.nix/a20de23b925fd8264fd7fad6454652e142fd7f73' (2022-08-14)
  → 'github:hercules-ci/gitignore.nix/637db329424fd7e46cf4185293b9cc8c88c95394' (2024-02-28)
• Updated input 'pre-commit-hooks/nixpkgs-stable':
    'github:NixOS/nixpkgs/c37ca420157f4abc31e26f436c1145f8951ff373' (2023-06-03)
  → 'github:NixOS/nixpkgs/614b4613980a522ba49f0d194531beddbb7220d3' (2024-03-17)
• Updated input 'rust-overlay':
    'github:oxalica/rust-overlay/e3ebc177291f5de627d6dfbac817b4a661b15d1c' (2023-11-17)
  → 'github:oxalica/rust-overlay/8eb8671512cb0c72c748058506e50c54fb5d8e2b' (2024-05-11)
• Updated input 'topiary':
    'github:tweag/topiary/79b93527d9bd59533f9a79fe490567963193fafd' (2023-10-23)
  → 'github:tweag/topiary/82a94a9f57104b89d4316afaeeab271cc51f7698' (2024-04-09)
• Updated input 'topiary/advisory-db':
    'github:rustsec/advisory-db/5ceeefcbbabf4b510ef8ede121d6dc57d1a1f7f8' (2023-07-08)
  → 'github:rustsec/advisory-db/0bc9a77248be5cb5f2b51fe6aba8ba451d74c6bb' (2024-04-03)
• Updated input 'topiary/crane':
    'github:ipetkov/crane/8b08e96c9af8c6e3a2b69af5a7fa168750fcf88e' (2023-07-07)
  → 'github:ipetkov/crane/06a9ff255c1681299a87191c2725d9d579f28b82' (2024-04-03)
• Removed input 'topiary/crane/flake-compat'
• Removed input 'topiary/crane/flake-utils'
• Removed input 'topiary/crane/flake-utils/systems'
• Removed input 'topiary/crane/rust-overlay'
• Removed input 'topiary/crane/rust-overlay/flake-utils'
• Removed input 'topiary/crane/rust-overlay/nixpkgs'
• Updated input 'topiary/flake-utils':
    'github:numtide/flake-utils/dbabf0ca0c0c4bce6ea5eaf65af5cb694d2082c7' (2023-06-25)
  → 'github:numtide/flake-utils/b1d9ab70662946ef0850d488da1c9019f3a9752a' (2024-03-11)
• Updated input 'topiary/nix-filter':
    'github:numtide/nix-filter/d90c75e8319d0dd9be67d933d8eb9d0894ec9174' (2023-06-19)
  → 'github:numtide/nix-filter/3342559a24e85fc164b295c3444e8a139924675b' (2024-03-11)
• Updated input 'topiary/rust-overlay':
    'github:oxalica/rust-overlay/d7181bb2237035df17cab9295c95f987f5c527e6' (2023-07-11)
  → 'github:oxalica/rust-overlay/20e7895d1873cc64c14a9f024a8e04f5824bed28' (2024-04-04)
• Updated input 'topiary/rust-overlay/flake-utils':
    'github:numtide/flake-utils/cfacdce06f30d2b68473a46042957675eebb3401' (2023-04-11)
  → 'github:numtide/flake-utils/1ef2e671c3b0c19053962c07dbda38332dcebf26' (2024-01-15)
• Updated input 'topiary/rust-overlay/nixpkgs':
    'github:NixOS/nixpkgs/96ba1c52e54e74c3197f4d43026b3f3d92e83ff9' (2023-04-13)
  → 'github:NixOS/nixpkgs/90f456026d284c22b3e3497be980b2e47d0b28ac' (2024-01-29)
```

### Running GitHub Actions on this PR

GitHub Actions will not run workflows on pull requests which are opened by a GitHub Action.

To run GitHub Actions workflows on this PR, run:

```sh
git branch -D update_flake_lock_action
git fetch origin
git checkout update_flake_lock_action
git commit --amend --no-edit
git push origin update_flake_lock_action --force
```

yannham commented:
@vkleen the `libcxxabi` seems to have disappeared from `llvmPackages`. Do you have any idea what to do about it? What was that used for? (it seems we are linking both `libcxx` and `libcxxabi` currently)

yannham commented:
I wondered if you knew what to do on the spot, but it's honestly not urgent, so this can wait if you have more urgent stuff to do.

jneem commented:
I checked that the static build still works if you just remove `cxxabi`. I am having some trouble with the nix coming from `nix-input`, though: after updating the flake, I get a weird error if I try to build anything:

```
❯ nix build
warning: Git tree '/home/jneeman/tweag/nickel' is dirty
error:
       … while fetching the input 'git+file:///home/jneeman/tweag/nickel'

       error: creating statement 'insert or replace into Cache(domain, key, value, timestamp) values (?, ?, ?, ?)': SQL logic error, table Cache has no column named domain (in '/home/jneeman/.cache/nix/fetcher-cache-v2.sqlite')
```

This error only happens with the `nix` in the dev shell; everything works if I use `/run/current-system/sw/bin/nix` instead. Is there some way to keep the unstable `nix` from leaking into the dev shell?

yannham commented:
I removed the mention, updated other inputs and made small fixes to get `nix flake check` to work, `nix develop` (thus fixing #1891 I believe), and `nix build .#nickel-static` with success.

I haven't tried `nix build` inside the devshell, but honestly, it's a bit of a strange thing to do. I don't know if it worked before, but I suppose you either build the project using `nix build` or you enter the devshell to use `cargo` and cos directly, but I don't think we should bother with trying to do a nixified build from within the devshell.

jneem commented on flake.nix:
```diff
@@ -154,6 +153,7 @@
 
           markdownlint = {
             enable = true;
+            # settings.configuration = builtins.fromJSON (builtins.readFile ./.markdownlint.json);
```
What is this comment for?

yannham replied:
Oops,  just an experimentation, because I thought markdownlint didn't take the `.markdownlint.json` file into account anymore. Turns out the schema changed, so the configuration just needed to be fixed.

jneem reviewed (commented):
Yeah, the nix in the devshell isn't a big deal. I'm in the devshell by default because I use direnv, but I can always leave it when needing to run nix commands.

yannham commented on flake.nix:
```diff
@@ -154,6 +153,7 @@
 
           markdownlint = {
             enable = true;
+            # settings.configuration = builtins.fromJSON (builtins.readFile ./.markdownlint.json);
```
```suggestion
```
